Mighty Flip Champs DX, N+ (nplus), EDGE and possibly other minis - audio speed problem

music plays so low (you have to turn your speakers up to full ) and so slow
rename txt to zip
[ppsspplog.txt](https://github.com/hrydgard/ppsspp/files/818358/ppsspplog.txt)

Sorry I ment [frame header](https://www.mp3-tech.org/programmer/frame_header.html), which each mp3 frame should have.

With QII, here's a bit from log:
```
HLE\sceIo.cpp:1018 stdout: mp3_start_play(./MP3/08.mp3) @ 0
HLE\sceIo.cpp:1018 stdout: seek 0: 0/10813602
\sceMp3.cpp:182 sceMp3ReserveMp3Handle(0bfcee74)
HLE\sceIo.cpp:1018 stdout: Playing ./MP3/08.mp3
\sceMp3.cpp:314 sceMp3Init(0bfcee74)
\sceMp3.cpp:332 sceMp3Init(): channels=2, samplerate=44100Hz, bitrate=256kbps
\sceMp3.cpp:425 sceMp3GetSamplingRate(0BFCEE74)
\sceMp3.cpp:401 sceMp3GetMp3ChannelNum(0BFCEE74)
\sceMp3.cpp:389 sceMp3SetLoopNum(0BFCEE74, 0)
MediaEngine.cpp:86 FF: Header missing
SimpleAudioDec.cpp:198 Error decoding Audio frame (32768 bytes): -1094995529 (bebbb1b7)
HLE\sceIo.cpp:1018 stdout: mp3_end_play
HLE\sceIo.cpp:1018 stdout: Clear res
```
it'll repeat constantly since game tries without end, with some variations depending on mp3, but I tried the most standard common ones, even tried stripped of any ID3 tags as well as rarer formats, at most it might play a few frames.

Searching for those in-game log prints I found this open source DQuakePlus so maybe it's [Mp3](https://github.com/st1x51/DQuakePlus/blob/99ded70e28d75859a4e68868470e7d4726a35f53/psp/mp3.c) code can make it easier to figure out, but it could as well use different variant of this code as I saw some variations of it in the net, the code was potentially buggy and fixed from homebrew to homebrew.

 I think mp3's in this homebrew shouldn't at least have the buffer "underfill" problem since it reserves a rather small pcm buffer size of 1024 and most common mp3 types would have 1152.

--- a/Core/Config.cpp
+++ b/Core/Config.cpp
@@ -673,7 +673,6 @@ static ConfigSetting soundSettings[] = {
 	ConfigSetting("AudioBackend", &g_Config.iAudioBackend, 0, true, true),
 	ConfigSetting("AudioLatency", &g_Config.iAudioLatency, 1, true, true),
 	ConfigSetting("ExtraAudioBuffering", &g_Config.bExtraAudioBuffering, false, true, false),
-	ConfigSetting("SoundSpeedHack", &g_Config.bSoundSpeedHack, false, true, true),
 	ConfigSetting("AudioResampler", &g_Config.bAudioResampler, true, true, true),
 	ConfigSetting("GlobalVolume", &g_Config.iGlobalVolume, VOLUME_MAX, true, true),
 

--- a/Core/Config.h
+++ b/Core/Config.h
@@ -81,7 +81,6 @@ public:
 	bool bPauseExitsEmulator;
 #endif
 	bool bPauseMenuExitsEmulator;
-	bool bPS3Controller;
 
 	// Core
 	bool bIgnoreBadMemAccess;

--- a/Core/Config.h
+++ b/Core/Config.h
@@ -201,9 +201,6 @@ public:
 	int iGlobalVolume;
 	bool bExtraAudioBuffering;  // For bluetooth
 
-	// Audio Hack
-	bool bSoundSpeedHack;
-
 	// UI
 	bool bShowDebuggerOnLoad;
 	int iShowFPSCounter;

--- a/Core/HW/SimpleAudioDec.cpp
+++ b/Core/HW/SimpleAudioDec.cpp
@@ -330,15 +330,8 @@ u32 AuCtx::AuDecode(u32 pcmAddr)
 	memset(outbuf, 0, PCMBufSize); // important! empty outbuf to avoid noise
 	u32 outpcmbufsize = 0;
 
-	int repeat = 1;
-	if (g_Config.bSoundSpeedHack){
-		repeat = 2;
-	}
-	int i = 0;
-	// decode frames in sourcebuff and output into PCMBuf (each time, we decode one or two frames)
-	// some games as Miku like one frame each time, some games like DOA like two frames each time
-	while (sourcebuff.size() > 0 && outpcmbufsize < PCMBufSize && i < repeat){
-		i++;
+	// decode frames in sourcebuff and output into PCMBuf, making sure it's filled
+	while (sourcebuff.size() > 0 && outpcmbufsize < PCMBufSize) {
 		int pcmframesize;
 		// decode
 		decoder->Decode((void*)sourcebuff.c_str(), (int)sourcebuff.size(), outbuf, &pcmframesize);
@@ -367,6 +360,7 @@ u32 AuCtx::AuDecode(u32 pcmAddr)
 		outbuf += pcmframesize;
 		// increase FrameNum count
 		FrameNum++;
+		break;
 	}
 	Memory::Write_U32(PCMBuf, pcmAddr);
 	return outpcmbufsize;

--- a/UI/GameSettingsScreen.cpp
+++ b/UI/GameSettingsScreen.cpp
@@ -514,9 +514,6 @@ void GameSettingsScreen::CreateViews() {
 		resampling->SetEnabledPtr(&g_Config.bEnableSound);
 	}
 
-	audioSettings->Add(new ItemHeader(a->T("Audio hacks")));
-	audioSettings->Add(new CheckBox(&g_Config.bSoundSpeedHack, a->T("Sound speed hack (DOA etc.)")));
-
 	// Control
 	ViewGroup *controlsSettingsScroll = new ScrollView(ORIENT_VERTICAL, new LinearLayoutParams(FILL_PARENT, FILL_PARENT));
 	controlsSettingsScroll->SetTag("GameSettingsControls");

--- a/UI/NativeApp.cpp
+++ b/UI/NativeApp.cpp
@@ -551,8 +551,6 @@ void NativeInit(int argc, const char *argv[], const char *savegame_dir, const ch
 					fileToLog = argv[i] + strlen("--log=");
 				if (!strncmp(argv[i], "--state=", strlen("--state=")) && strlen(argv[i]) > strlen("--state="))
 					stateToLoad = argv[i] + strlen("--state=");
-				if (!strncmp(argv[1], "--PS3", strlen("--PS3")))
-					g_Config.bPS3Controller = true;
 #if !defined(MOBILE_DEVICE)
 				if (!strncmp(argv[i], "--escape-exit", strlen("--escape-exit")))
 					g_Config.bPauseExitsEmulator = true;

--- a/UI/SavedataScreen.cpp
+++ b/UI/SavedataScreen.cpp
@@ -339,8 +339,6 @@ bool SavedataBrowser::ByFilename(const UI::View *v1, const UI::View *v2) {
 
 static time_t GetTotalSize(const SavedataButton *b) {
 	auto fileLoader = std::unique_ptr<FileLoader>(ConstructFileLoader(b->GamePath()));
-	tm datetm;
-	bool success;
 	switch (Identify_File(fileLoader.get())) {
 	case IdentifiedFileType::PSP_PBP_DIRECTORY:
 	case IdentifiedFileType::PSP_SAVEDATA_DIRECTORY:

--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -185,7 +185,6 @@ static RetroOption<bool> ppsspp_gpu_hardware_transform("ppsspp_gpu_hardware_tran
 static RetroOption<bool> ppsspp_vertex_cache("ppsspp_vertex_cache", "Vertex Cache (Speedhack)", true);
 static RetroOption<bool> ppsspp_separate_io_thread("ppsspp_separate_io_thread", "IO Threading", false);
 static RetroOption<bool> ppsspp_unsafe_func_replacements("ppsspp_unsafe_func_replacements", "Unsafe FuncReplacements", true);
-static RetroOption<bool> ppsspp_sound_speedhack("ppsspp_sound_speedhack", "Sound Speedhack", false);
 static RetroOption<bool> ppsspp_cheats("ppsspp_cheats", "Internal Cheats Support", false);
 static RetroOption<IOTimingMethods> ppsspp_io_timing_method("ppsspp_io_timing_method", "IO Timing Method", { { "Fast", IOTimingMethods::IOTIMING_FAST }, { "Host", IOTimingMethods::IOTIMING_HOST }, { "Simulate UMD delays", IOTimingMethods::IOTIMING_REALISTIC } });
 
@@ -213,7 +212,6 @@ void retro_set_environment(retro_environment_t cb) {
 	vars.push_back(ppsspp_vertex_cache.GetOptions());
 	vars.push_back(ppsspp_separate_io_thread.GetOptions());
 	vars.push_back(ppsspp_unsafe_func_replacements.GetOptions());
-	vars.push_back(ppsspp_sound_speedhack.GetOptions());
 	vars.push_back(ppsspp_cheats.GetOptions());
 	vars.push_back(ppsspp_io_timing_method.GetOptions());
 	vars.push_back({});
@@ -278,7 +276,6 @@ static void check_variables(CoreParameter &coreParam) {
 	ppsspp_texture_replacement.Update(&g_Config.bReplaceTextures);
 	ppsspp_separate_io_thread.Update(&g_Config.bSeparateIOThread);
 	ppsspp_unsafe_func_replacements.Update(&g_Config.bFuncReplacements);
-	ppsspp_sound_speedhack.Update(&g_Config.bSoundSpeedHack);
 	ppsspp_cheats.Update(&g_Config.bEnableCheats);
 	ppsspp_locked_cpu_speed.Update(&g_Config.iLockedCPUSpeed);
 	ppsspp_rendering_mode.Update(&g_Config.iRenderingMode);

